lib/AST/Decl.cpp

```diff
@@ -1170,12 +1170,26 @@ SourceRange PatternBindingEntry::getOrigInitRange() const {
 }
 
 void PatternBindingEntry::setInit(Expr *E) {
+  auto PreviousInit = getInit();
   auto F = InitAndFlags.getInt();
   if (E) {
     InitAndFlags.setInt(F - Flags::Removed);
     InitAndFlags.setPointer(E);
+    if (!PreviousInit) {
```

Comment:
Is there no better place for this, maybe in Sema?

Reply:
These initialisers get set and unset all over the place. Are you thinking that the attribute should be added once, after all of that?

Reply:
It does seem like there's a lot of variables created and not in several places throughout the compiler and REPL, etc., and it seems not all of them go through the "obvious" type checker paths (e.g. property behavior's `completePropertyBehaviorStorage` sets everything up perfectly, and skips the main checker). It seems unfortunate to push this detail to be repeated at all such code paths (possibly including in lldb), given it is entirely computable from info given to this call, but I agree that it's a little ugly to have it here.

Reply:
I've moved it to `TypeChecker::typeCheckPatternBinding`.